src/cuarray.jl

```diff
@@ -88,3 +89,33 @@ end
 #     KnetArray{T,N}(k, size(x))
 # end
 
+
+# Testing the CuArrays allocator: set Knet.cuallocator()=true to use this
+function KnetPtrCu(len::Int)
+    c = CuArray{UInt8}(undef, len)
+    p = convert(Cptr, convert(Int, c.buf.ptr))
+    KnetPtr(p, len, gpu(), c)
+end
+
+function freeKnetPtrCu(p::KnetPtr)
+    if p.parent isa CuArray
+        unsafe_free!(p.parent)
+        p.ptr = C_NULL
+        p.parent = nothing
+    else
+        error("Bad parent pointer $(p.parent)")
+    end
+end
+
+# argmax, argmin etc. Fixes https://github.com/denizyuret/Knet.jl/issues/368.
+# Two options: argmax(Array(KnetArray)) vs argmax(CuArray)
+# Experiments: 10x10, 100x100, 1000x1000 with no dims, dims=1, dims=2
+# With no dims, CuArrays is better for 100x100, 1000x1000.
+# With all others, KnetArray is better.
+
+import Base: argmax, argmin, findmax, findmin
+argmaxarray(x,d)=((d===:) && length(x) > 4096 ? CuArray(x) : Array(x))
```

Comment:
Given issue https://github.com/JuliaGPU/CuArrays.jl/issues/304, I think it is better to just use `Array(x)` for all cases

Reply:
Thanks, good catch. I will switch to Arrays until the issue is fixed.